config/src/lib.rs

```diff
@@ -568,7 +574,7 @@ mod tests {
             OsString::from("--testmultiple"),
             OsString::from("3"),
         ];
-        let config: TestConfig = ConfigBuilder::build(command_line_args, &env_var_name)
+        let config: TestConfig = *ConfigBuilder::build(command_line_args, &env_var_name)
```

Comment:
All of these derefs can go if you make config type Box<TestConfig>, it'll be dereferenced automatically later so it should compile.

That said: See commment about Box earlier.